[FLINK-18659][hive][orc] Fix streaming write for Hive 1.x Orc table

lirui-apache opened the pull request:
<!--
*Thank you very much for contributing to Apache Flink - we are happy that you want to help us improve Flink. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*

*Please understand that we do not do this to make contributions to Flink a hassle. In order to uphold a high standard of quality for code contributions, while at the same time managing a large number of contributions, we need contributors to prepare the contributions well, and give reviewers enough contextual information for the review. Please also understand that contributions that do not follow this guide will take longer to review and thus typically be picked up with lower priority by the community.*

## Contribution Checklist

  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/FLINK/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
  
  - Name the pull request in the form "[FLINK-XXXX] [component] Title of the pull request", where *FLINK-XXXX* should be replaced by the actual issue number. Skip *component* if you are unsure about which is the best component.
  Typo fixes that have no associated JIRA issue should be named following this pattern: `[hotfix] [docs] Fix typo in event time introduction` or `[hotfix] [javadocs] Expand JavaDoc for PuncuatedWatermarkGenerator`.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
  
  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` passes. You can set up Azure Pipelines CI to do that following [this guide](https://cwiki.apache.org/confluence/display/FLINK/Azure+Pipelines#AzurePipelines-Tutorial:SettingupAzurePipelinesforaforkoftheFlinkrepository).

  - Each pull request should address only one issue, not mix up code from multiple issues.
  
  - Each commit in the pull request has a meaningful commit message (including the JIRA id)

  - Once all items of the checklist are addressed, remove the above text and this checklist, leaving only the filled out template below.


**(The sections below can be removed for hotfixes of typos)**
-->

## What is the purpose of the change

Fix streaming write to Hive 1.x Orc tables.


## Brief change log

  - Check whether file exists before calling `FileSystem::getFileStatus` to get the size of in-progress files.
  - Fix serialization issue for `OrcNoHiveBulkWriterFactory`
  - Add `NoHivePhysicalWriterImpl` to fix NoSuchMethodError due to protobuf relocation in `orc-core-nohive`
  - Add tests for streaming write to orc table.


## Verifying this change

Existing and added tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? NA

lirui-apache commented:
@JingsongLi Pleas help review this PR, thanks!

JingsongLi commented:
You can remove `hive.connector.orc.version`, and use `orc.version` in `flink-sql-connector-hive...`

JingsongLi commented:
Please create PR for 1.11.

lirui-apache commented:
As discussed offline, I have renamed the property name for the orc version needed by Hive 2.x. Also updated `flink-sql-connector-hive-1.2.2` to use the new orc version.

JingsongLi commented on flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableSinkITCase.java:
```diff
@@ -252,32 +252,45 @@ public void testBatchAppend() {
-		testStreamingWrite(true, false, false, this::checkSuccessFiles);
+		testStreamingWrite(true, false, "parquet", this::checkSuccessFiles);
+		// disable orc test for hive 2.x due to dependency conflict
+		if (!hiveCatalog.getHiveVersion().startsWith("2.")) {
```
Why dependency conflict? Can we make it correct?

lirui-apache commented:
@JingsongLi I have enabled mr orc writer tests for 2.x
